packages/client/components/ActivityLibrary/ActivityDetails.tsx

```diff
@@ -108,24 +121,63 @@ const ActivityDetails = (props: Props) => {
 
   const removeTemplate = useCallback(() => {
     if (submitting) return
-    submitMutation()
-    RemoveReflectTemplateMutation(
-      atmosphere,
-      {templateId},
-      {
-        onError,
-        onCompleted: () => {
-          onCompleted()
-          history.replace('/activity-library')
+    if (selectedTemplate?.type === 'retrospective') {
+      submitMutation()
+      RemoveReflectTemplateMutation(
+        atmosphere,
+        {templateId},
+        {
+          onError,
+          onCompleted: () => {
+            onCompleted()
+            history.replace('/activity-library')
+          }
         }
-      }
-    )
+      )
+    } else if (selectedTemplate?.type === 'poker') {
+      submitMutation()
+      RemovePokerTemplateMutation(
+        atmosphere,
+        {templateId},
+        {
+          onError,
+          onCompleted: () => {
+            onCompleted()
+            history.replace('/activity-library')
+          }
+        }
+      )
+    }
   }, [templateId, submitting, submitMutation, onError, onCompleted])
 
-  const {togglePortal, modalPortal, closePortal} = useModal({
+  const {
+    togglePortal: toggleTeamPickerPortal,
+    modalPortal: teamPickerModalPortal,
+    closePortal: closeTeamPickerPortal
+  } = useModal({
     id: 'templateTeamPickerModal'
   })
 
+  const {
+    openPortal: openPokerTemplateScaleDetailsPortal,
+    modalPortal: pokerTemplateScaleDetailsPortal,
+    closePortal: closePokerTemplateScaleDetailsPortal
+  } = useModal({
+    id: 'pokerTemplateScaleDetailsModal'
+  })
+
+  useEffect(() => {
+    if (selectedTemplate?.team.editingScaleId) {
```

Comment:
It might be nice to keep track of `editingScaleId` in component state instead of global client-side relay state, but a refactor for this seems premature, so I'm simply reusing the existing global state + logic for simplicity. We should consider this refactor if/when we fully adopt the activity library.